etc/kayobe/ansible/get-cloud-facts.yml

```
@@ -0,0 +1,124 @@
+---
+- name: Get facts for control host
+  hosts: localhost
+  gather_facts: true
+  tasks:
+    - name: Check if ngs enabled
+      ansible.builtin.set_fact:
+        ngs_enabled: >-
+          {{ 'genericswitch' in kolla_neutron_ml2_mechanism_drivers | default(false) }}
+      when: kolla_neutron_ml2_mechanism_drivers
+
+- name: Get facts for seed node
+  hosts: seed
+  gather_facts: true
+  tasks:
+    - name: Is there a dedicated seed node?
+      ansible.builtin.set_fact:
+        seed_node_is_vm: "{{ ansible_facts.virtualization_role == 'guest' }}"
+
+- name: Get facts
+  hosts: all
+  gather_facts: true
+  tasks:
+    - name: Pinging stackhpc.com to check internet connectivity
+      ansible.builtin.command: "ping stackhpc.com -c 3"
+      register: successful_ping
+      timeout: 5
+      ignore_errors: true
+
+    - name: Set internet connectivity fact
+      ansible.builtin.set_fact:
+        internet_connectivity: "{{ not successful_ping.failed }}"
+    - name: Get OS distribution
+      ansible.builtin.set_fact:
+        distribution: "{{ ansible_facts.distribution }}"
+
+
+    - name: Get OS distribution release
+      ansible.builtin.set_fact:
+        distribution_release: "{{ ansible_facts.distribution_release }}"
```

I went looking for ways to deduplicate this and it turns out you can specify multiple facts in one task, see [docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/set_fact_module.html)

So you can do:
```suggestion
    - name: Set facts
      ansible.builtin.set_fact:
        internet_connectivity: "{{ not successful_ping.failed }}"
        distribution: "{{ ansible_facts.distribution }}"
        distribution_release: "{{ ansible_facts.distribution_release }}"
        hugepages_enabled: "{{ 'hugepages' in ansible_facts.cmdline }}"
```
(note I've also added `hugepages_enabled` to my example)

---

This looks a lot tidier thanks